gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/ExecutionTreeBuilder.java

```diff
@@ -78,12 +85,17 @@ public QueryNode build() {
     // the data store
     if (singleSourceForAllAttributes.isPresent()) {
```

Comment:
Once we have the optimization of pushing the pagination down if the order by and filter are on the same source, we can actually get rid of the entire single source check